tests/test_main.py

```diff
@@ -114,6 +114,19 @@ def mock_responses(*args, **kwargs):
     vault_dump.main.get_pki_roles(".", "token", "addr", "mount_path/")
 
 
+@pytest.mark.parametrize("list_status_code, get_status_code", [pytest.param(403, 200), pytest.param(200, 403), pytest.param(200, 200)])
+def test_get_pki_certs(mock_request, mock_shutil, mock_path, list_status_code, get_status_code):
+    def mock_responses(*args, **kwargs):
+        if args[-1].endswith("roles"):
+            return mock.Mock(status_code=list_status_code, json=mock.Mock(return_value={"data":{"keys": ['role1', 'role2'],}}))
+        elif "roles/" in args[-1]:
+            return mock.Mock(status_code=get_status_code, json=mock.Mock(return_value={"data":{}}))
+        else:
+            return mock.Mock(json=mock.Mock(return_value={}))
+    mock_request.side_effect = mock_responses
+    vault_dump.main.get_pki_certs(".", "token", "addr", "mount_path/")
+
+
```

Comment:
nit: To follow-suit with other testing patterns, line 149 should also mock out `get_pki_certs` so that it's only tested in isolation